utils/sample_qc.py

```diff
@@ -357,6 +357,102 @@ def infer_sex(
     return (sex_ht)
 
 
+def get_sex_expr(
+    chr_x_ploidy: hl.expr.NumericExpression,
+    chr_y_ploidy: hl.expr.NumericExpression,
+    f_stat: hl.expr.NumericExpression,
+    xx_x_ploidy_cutoffs: Tuple[float, float] = (1.4, 2.25),
+    xy_x_ploidy_cutoffs: Tuple[float, float] = (0.5, 1.4),
+    xx_y_ploidy_cutoff: float = 0.1,
+    xy_y_ploidy_cutoffs: Tuple[float, float] = (0.15, 1.2),
+    yy_y_ploidy_cutoff: float = 1.3,
+    xxx_x_ploidy_cutoff:float = 2.5,
+    f_stat_female_cutoff: float = -0.2,
+    f_stat_male_cutoff: float = 0.2
+) -> hl.expr.StructExpression:
+    # TODO: Automate cutoffs
+    # TODO: Provide alternatives in case of missing annotations (e.g. no chr_y_ploidy)
+    """
+    Creates a struct with the following annotations:
+    - karyoptype (str)
+    - sex (str), which can be either 'male', 'female' or missing
+    - is_female (missing for karyotypes that aren't either 'XX' or 'XY')
+
+    :param NumericExpression chr_x_ploidy: chrom X ploidy (or relative ploidy)
+    :param NumericExpression  chr_y_ploidy: chrom Y ploidy (or relative ploidy)
+    :param NumericExpression f_stat: chrom X F-stat
+    :param Tuple[float, float] xx_x_ploidy_cutoffs: Boundaries around the chom X ploidy for females
```

Comment:
So, I think we should have functionality to automatically compute these boundaries. While the min value is 0 essentially everywhere, all ploidy cutoffs can exceed 1 I think.
I honestly think that anyone who's going to enter a negative value for ploidy should not be using this function 😂

Reply:
haha sorry, my comment was not clear. I meant more documenting that the min threshold should be in the 0-index in the tuple, while the max threshold should be in the 1-index position of the tuple. After thinking about it though, I think everyone would put [min,max] instead of [max,min] so we don't need to document it.